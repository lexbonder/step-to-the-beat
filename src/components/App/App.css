/* Colors
Dark Green = #4E9AF5
Light Gray = #999999
Medium Gray = #404040
Dark Gray = #262626
*/

* {
  box-sizing: border-box;
  outline: none;
}

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  background-color: #262626;
}

main {
  margin-top: 80px;
}

.App {
  max-width: 700px;
  margin: auto;
}

h1, h2, h3, h4, p {
  margin: 0;
}

h4 {
  color: #262626;
}

a {
  text-decoration: none;
}

p {
  color: #999999;
}

.show {
  display: block;
}

.hide {
  display: none;
}

.error {
  text-align: center;
  color: black;
  background-color: #FF4A19;
}

@media (min-width: 351px) {
  .playlist {
    margin-top: 195px;
  }
}

@media (min-width: 500px) {
  .Home {
    height: 50vh;
    margin-top: 200px;
  }

  .intro {
    max-width: 650px;
  }

  .SelectSpm {
    max-width: 550px;
  }

  .SelectGenre {
    max-width: 550px;
    margin: 30px auto;
  }

  .button-wrapper {
    max-width: 550px;
    margin: auto;
  }

  .bottom-half-wrapper {
    display: flex;
    justify-content: space-around;
  }

  .new-spm-button {
    margin: 70px auto;
  }
}

@media (min-width: 553px) {
  .playlist {
    margin-top: 170px;
  }
}

@media (min-width: 700px) {
  header,
  .playlist-header {
    width: 700px;
  }

}