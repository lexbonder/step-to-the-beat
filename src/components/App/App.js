import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import Header from '../../containers/Header/Header';
import Home from '../../containers/Home/Home';
import Login from '../../containers/Login/Login';
import SelectGenre from '../../containers/SelectGenre/SelectGenre';
import Calculations from '../Calculations/Calculations';
import { SavedSpms } from '../../containers/SavedSpms/SavedSpms';
import { FavoriteSongs } from '../../containers/FavoriteSongs/FavoriteSongs';
import { FavoritePlaylists } from '../../containers/FavoritePlaylists/FavoritePlaylists';
import { SelectSPM } from '../../containers/SelectSPM/SelectSPM';
import { Confirm } from '../../containers/Confirm/Confirm';
import './App.css';

export class App extends Component {

  render() {
    return (
      <div>
        <Route path='/' component={Header} />
        <Route path='/' component={Home} /> 
        
        <Route exact path='/calculations' component={Calculations} />
        <Route exact path='/saved-spms' component={SavedSpms}/>
        <Route exact path='/favorite-songs' component={FavoriteSongs} />
        <Route exact path='/favorite-playlists' component={FavoritePlaylists} />
        <Route exact path='/select-spm' component={SelectSPM} />
        <Route exact path='/select-genre' component={SelectGenre} />
        <Route exact path='/confirm' component={Confirm} />
        
        <Route exact path='/login' component={Login} />
      </div>
    )
  }
}