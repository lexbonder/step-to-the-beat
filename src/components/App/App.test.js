import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';

it('should match the snapshot', () => {
  const wrapper = shallow(<App />);

  expect(wrapper).toMatchSnapshot()
});
