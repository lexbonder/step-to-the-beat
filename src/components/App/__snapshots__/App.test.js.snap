// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/calculations"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/saved-playlists"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/select-spm"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/select-genre"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/confirm"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/playlist"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/login"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/calculations",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/saved-playlists",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/select-spm",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/select-genre",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/confirm",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/playlist",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/calculations"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/saved-playlists"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/select-spm"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/select-genre"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/confirm"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/playlist"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/login"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/calculations",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/saved-playlists",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/select-spm",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/select-genre",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/confirm",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/playlist",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
